fpmsyncd/routesync.cpp

```diff
@@ -886,6 +1004,105 @@ bool RouteSync::getIfName(int if_index, char *if_name, size_t name_len)
     return true;
 }
 
+/*
+ * getNextHopList() - parses next hop list attached to route_obj
+ * @arg route_obj     (input) Netlink route object
+ * @arg gw_list       (output) comma-separated list of NH IP gateways
+ * @arg mpls_list     (output) comma-separated list of NH MPLS info
+ * @arg intf_list     (output) comma-separated list of NH interfaces
+ *
+ * Return bool true=Success false=Failure
+ */
+bool RouteSync::getNextHopList(struct rtnl_route *route_obj, string& gw_list,
+                               string& mpls_list, string& intf_list)
+{
+    bool mpls_found = false;
+
+    for (int i = 0; i < rtnl_route_get_nnexthops(route_obj); i++)
+    {
+        struct rtnl_nexthop *nexthop = rtnl_route_nexthop_n(route_obj, i);
+        struct nl_addr *addr = NULL;
+
+        /* RTA_GATEWAY is NH gateway info for IP routes only */
+        if ((addr = rtnl_route_nh_get_gateway(nexthop)))
+        {
+            char gw_ip[MAX_ADDR_SIZE + 1] = {0};
+            nl_addr2str(addr, gw_ip, MAX_ADDR_SIZE);
+            gw_list += gw_ip;
+        }
+        /* RTA_VIA is NH gateway info for MPLS routes only */
+        else if ((addr = rtnl_route_nh_get_via(nexthop)))
+        {
+            char via_ip[MAX_ADDR_SIZE + 1] = {0};
+            nl_addr2str(addr, via_ip, MAX_ADDR_SIZE);
+            gw_list += via_ip;
+        }
+        else
+        {
+            if (rtnl_route_get_family(route_obj) == AF_INET6)
+            {
+                gw_list += "::";
+            }
+            /* for MPLS route, use IPv4 as default gateway. */
+            else
+            {
+                gw_list += "0.0.0.0";
+            }
+        }
+
+        /* RTA_NEWDST is MPLS NH label stack for MPLS routes only */
+        if ((addr = rtnl_route_nh_get_newdst(nexthop)))
+        {
+            char label[MAX_ADDR_SIZE + 1] = {0};
+            nl_addr2str(addr, label, MAX_ADDR_SIZE);
+            mpls_list += string("push+");
+            mpls_list += label;
+            mpls_found = true;
+        }
+        /* LWTUNNEL_ENCAP_MPLS RTA_DST is MPLS NH label stack for IP routes only */
+        else if ((addr = rtnl_route_nh_get_encap_mpls_dst(nexthop)))
```

Comment:
can we make combine two if condition so that code remain common

Reply:
Done.  Function is updated to have only one condition where it was previously repeated.